src/cli/vector_search.rs

```diff
@@ -85,19 +93,88 @@ fn run(
     engine_state: &EngineState,
     stack: &mut Stack,
     call: &Call,
-    _input: PipelineData,
+    input: PipelineData,
 ) -> Result<PipelineData, ShellError> {
     let span = call.head;
     let ctrl_c = engine_state.ctrlc.as_ref().unwrap().clone();
 
+    let mut vector: Vec<f32> = vec![];
+    match input.into_value(span) {
+        Value::List { vals, span: _ } => {
+            let rec = match vals[0].as_record() {
+                Ok(r) => r,
+                Err(e) => {
+                    return Err(ShellError::GenericError(
+                        "Please supply vector or output from `vector enrich-text`".to_string(),
+                        "".to_string(),
+                        None,
+                        None,
+                        vec![e],
+                    ));
+                }
+            };
+
+            if rec.0 == vec!["id", "content"] {
```

Comment:
I'm slightly suspicious of this

Reply:
I am also, will have a think about how to improve